android/src/main/java/com/pspdfkit/react/events/PdfViewDataReturnedEvent.java

```diff
@@ -0,0 +1,67 @@
+package com.pspdfkit.react.events;
+
+import android.support.annotation.IdRes;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.uimanager.events.Event;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+import com.pspdfkit.annotations.Annotation;
+import com.pspdfkit.react.helper.JsonUtilities;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PdfViewDataReturnedEvent extends Event<PdfViewDataReturnedEvent> {
+
+    public static final String EVENT_NAME = "pdfViewDataReturned";
+
+    private final WritableMap payload;
+
+    public PdfViewDataReturnedEvent(@IdRes int viewId, int requestId, List<Annotation> annotations) {
+        super(viewId);
+        Map<String, Object> map = new HashMap<>();
+        map.put("requestId", requestId);
+        try {
+            List<Map<String, Object>> annotationsSerialized = new ArrayList<>();
+            for (Annotation annotation : annotations) {
+                JSONObject instantJson = new JSONObject(annotation.toInstantJson());
```

Comment:
I followed the discussion about not being able to convert detached annotations to JSON. Could this happen here too? If so, we probably need handling at this point.

Reply:
This shouldn't happen since this is only called from the main thread as a result of listing annotation of a page.